src/components/Form.tsx

```diff
@@ -0,0 +1,155 @@
+        console.log({ entryInfo });
+        if (entryInfo.data) {
+          setSelections({ FLOW: entryInfo.data.flow, MOOD: entryInfo.data.mood, CRAVINGS: entryInfo.data.craving });
+          setSymptoms(entryInfo.data.symptom ?? '');
+        } else {
+          setSelections({ FLOW: null, MOOD: null, CRAVINGS: null });
+          setSymptoms('');
+        }
+      } catch (error) {
+        if (error instanceof Error) setError(error);
+      }
+      setLoading(false);
+    };
+    getFormData();
+
+    return () => setError(null);
+  }, [isAuthorized, entryDate]);
+
+  const goToDate = (num: number) => {
+    const newDate = new Date(entryDate);
+    newDate.setDate(newDate.getDate() + num);
+    updateEntryDate(newDate);
+  };
+
+  const postForm = async () => {
+    const infoOptions = [...Object.values(selections), symptoms];
+    try {
+      if (infoOptions.every((option) => !option)) {
+        throw new Error('Please input something to save!');
+      }
+      const postedRes = await postEntry(router.asPath.includes('demo'), 'addEntry', {
+        flow: selections.FLOW,
+        craving: selections.CRAVINGS,
+        mood: selections.MOOD,
+        symptom: symptoms,
+        user_id: data ? data[0].passage_user_id : '',
+        date: `${new Date(entryDate).getFullYear()}-${new Date(entryDate).getMonth() + 1}-${new Date(entryDate).getDate()}`,
+      });
+      setError(null);
+      router.push(`${router.asPath.includes('demo') ? '/demo' : ''}/insights`);
+    } catch (error) {
+      if (error instanceof Error) setError(error);
+    }
+  };
+
+  const updateSelections = (title: string, option: string) => {
+    const newOption = selections[title] === option ? null : option;
+    setSelections((prev) => ({ ...prev, [title]: newOption }));
+  };
+
+  const formEls = formSections.map((section) => {
+    let allOptions = section.options.map((option) => {
+      return (
+        <button key={option} onClick={() => updateSelections(section.title, option)} className='m-5 w-14 flex justify-content flex-col items-center'>
+          <div className={`${selections[section.title] === option ? `${styles['selected-option']}` : 'bg-white light-opacity-bg'}` + ' rounded-full h-14 w-14 flex justify-center items-center'}>
+            <Image width={64} height={64} className={`${option === 'Confident' ? 'rounded-bl-xl' : ''}` + ' w-5/6 h-5/6'} src={`/images/FormIcons/${option}.png`} alt={option} />
+          </div>
+          <p className='min-w-max text-white thin-regular'>{option}</p>
+        </button>
+      );
+    });
+
+    return (
+      <div key={section.title}>
+        <h3 className='ml-2 text-white thick-regular'>{section.title}</h3>
+        <div className={`${styles['scroll-area-no-track']}` + ' flex max-w-100vw overflow-x-auto justify-between'}>{allOptions}</div>
+      </div>
+    );
+  });
+
+  return (
```

Comment:
I think another potential refactor could be getting our form to be more semantic by being in a form element. I def understand how that would be tricky with our form being just a bunch of buttons though.